engine/src/main/java/org/hibernate/validator/internal/metadata/location/TypeArgumentConstraintLocation.java

```diff
@@ -114,6 +117,7 @@ public boolean equals(Object o) {
 	@Override
 	public int hashCode() {
 		int result = delegate != null ? delegate.hashCode() : 0;
+		result = 31 * result + typeParameter.hashCode();
 		result = 31 * result + typeForValidatorResolution.hashCode();
```

Comment:
Do we actually need `typeForValidatorResolution` in here? Shouldn't delegate and type parameter suffice for determining the identity?